### PR TITLE
Update tests to match the testing guidelines

### PR DESCRIPTION
**Pull request by gasperzgonec**
## Summary

Updated tests to match the testing guidelines specified inside of CONTRIBUTING.md file.
I have changed how the testing works.

Before, we had `npm run test`, which ran all tests, including timeout tests, which took quite a while. I've removed them from the default testing and removed the coverage as well.

Full and coverage tests include timeout tests and can be accessed using `test:full` and `test:cov`. The actions should use `test:cov` by default and work as before.
This speeds up testing a lot (3.5s vs 26s).

## Connected Issues

- [#ISS-202840](https://app.devrev.ai/devrev/works/ISS-202840) ([Needs to be merged before this, as it uses updated CONTRIBUTING.md](https://github.com/devrev/adaas-sdk/pull/61))
- [#ISS-202839](https://app.devrev.ai/devrev/works/ISS-202839)

### CODE REVIEW

File: src/uploader/uploader.test.ts [hunk 1]
```diff
@@ -71,49 +71,51 @@ describe('Uploader Class Tests', () => {
...
-      getArtifactUploadUrlMockResponse
-    );
-    // Mock unsuccessful response for uploadArtifact
-    (axiosClient.post as jest.Mock).mockResolvedValueOnce(undefined);
-
-    const entity = 'entity';
-    const fetchedObjects = [{ key: 'value' }];
-    const uploadResponse = await uploader.upload(entity, fetchedObjects);
+      expect(uploadResponse.error).toBeInstanceOf(Error);
+      expect(uploadResponse.error?.message).toBe('Error while getting artifact upload URL.');
```
Comment (patricijabrecko) [inline src/uploader/uploader.test.ts:84]: Here also let's remove text comparison

File: src/uploader/uploader.test.ts [hunk 2]
```diff
@@ -71,49 +71,51 @@ describe('Uploader Class Tests', () => {
...
-    (axiosClient.post as jest.Mock).mockResolvedValueOnce(undefined);
+      const entity = 'entity';
+      const fetchedObjects = [{ key: 'value' }];
+      const uploadResponse = await uploader.upload(entity, fetchedObjects);
 
-    const entity = 'entity';
-    const fetchedObjects = [{ key: 'value' }];
-    const uploadResponse = await uploader.upload(entity, fetchedObjects);
+      expect(uploadResponse.error).toBeInstanceOf(Error);
+      expect(uploadResponse.error?.message).toBe('Error while uploading artifact.');
```
Comment (patricijabrecko) [inline src/uploader/uploader.test.ts:100]: And here

File: src/uploader/uploader.test.ts [hunk 3]
```diff
@@ -71,49 +71,51 @@ describe('Uploader Class Tests', () => {
...
+      (axiosClient.post as jest.Mock).mockResolvedValueOnce(getSuccessResponse());
+      // Mock unsuccessful response from confirmArtifactUpload
+      (axiosClient.post as jest.Mock).mockResolvedValueOnce(undefined);
+
+      const entity = 'entity';
+      const fetchedObjects = [{ key: 'value' }];
+      const uploadResponse = await uploader.upload(entity, fetchedObjects);
+
+      expect(uploadResponse.error).toBeInstanceOf(Error);
+      expect(uploadResponse.error?.message).toBe('Error while confirming artifact upload.');
```
Comment (patricijabrecko) [inline src/uploader/uploader.test.ts:118]: And here

File: src/workers/worker-adapter.test.ts [hunk 4]
```diff
@@ -121,188 +121,190 @@ describe('WorkerAdapter', () => {
...
-    
-      // Mock getting attachments
-      adapter['uploader'].getAttachmentsFromArtifactId = jest.fn().mockResolvedValue({
-        attachments: [
-          { url: 'http://example.com/file1.pdf', id: 'attachment1', file_name: 'file1.pdf', parent_id: 'parent1' },
-        ],
+        // Assert
+        expect(consoleWarnSpy).toHaveBeenCalledWith(
+          'The specified batch size (0) is invalid. Using 1 instead.'
+        );
```
Comment (patricijabrecko) [inline src/workers/worker-adapter.test.ts:157]: Let's remove all these console spies (unless it is the only way to test, but I think it shouldn't be anywhere)

File: src/attachments-streaming/attachments-streaming-pool.test.ts [hunk 5]
```diff
@@ -274,7 +274,7 @@ describe('AttachmentsStreamingPool', () => {
     });
   });
 
-  describe('edge cases', () => {
+  describe('[Edges]', () => {
```
[the remark this answers is not in the document]
Comment (gasperzgonec) [reply src/attachments-streaming/attachments-streaming-pool.test.ts:277]: Nice catch!

File: src/common/install-initial-domain-mapping.test.ts [hunk 6]
```diff
@@ -193,58 +193,60 @@ describe('installInitialDomainMapping', () => {
...
 
-    mockAxiosClient.get.mockResolvedValueOnce(snapInResponseWithoutSlug);
+      mockAxiosClient.get.mockResolvedValueOnce(snapInResponseWithoutSlug);
 
-    await expect(
-      installInitialDomainMapping(mockEvent, mockInitialDomainMapping)
-    ).rejects.toThrow('No import slug or snap-in slug found');
+      await expect(
+        installInitialDomainMapping(mockEvent, mockInitialDomainMapping)
+      ).rejects.toThrow('No import slug or snap-in slug found');
```
[the remark this answers is not in the document]
Comment (gasperzgonec) [reply src/common/install-initial-domain-mapping.test.ts]: I have removed all string comparisons from the tests.

File: src/workers/worker-adapter.test.ts (hunk 4 above)
Comment (patricijabrecko) [reply src/workers/worker-adapter.test.ts]: Makes sense!

File: jest.config.cjs [hunk 7]
```diff
@@ -1,4 +1,8 @@
 module.exports = {
   preset: 'ts-jest',
-  testPathIgnorePatterns: ['./dist/'],
+  testPathIgnorePatterns: [
+    './dist/',
+    // Exclude timeout tests by default - they should only run with test:full or test:cov
+    './src/tests/timeout-handling/'
```
[the remark this answers is not in the document]
Comment (gasperzgonec) [reply jest.config.cjs:6]: The difference is 13s to 32s, so it's not a major degrade.
I'll revert the test split.

File: package.json [hunk 8]
```diff
@@ -11,7 +11,9 @@
     "start": "ts-node src/index.ts",
     "lint": "eslint .",
     "lint:fix": "eslint .  --fix",
-    "test": "jest --forceExit --coverage --runInBand"
+    "test": "jest --silent --verbose=false",
+    "test:full": "jest --forceExit --runInBand --testPathIgnorePatterns='./dist/' --silent --verbose=false",
+    "test:cov": "jest --forceExit --coverage --runInBand --testPathIgnorePatterns='./dist/' --silent --verbose=false"
```
[the remark this answers is not in the document]
Comment (gasperzgonec) [reply package.json:16]: That is true.
I'm used to running ultra-fast checks that verify the behaviour and running the long tests in git's pre-commit hook.

But you are indeed correct and ~20s increase in time is not worth complexity increase.

File: src/attachments-streaming/attachments-streaming-pool.test.ts [hunk 9]
```diff
@@ -66,7 +66,7 @@ describe('AttachmentsStreamingPool', () => {
     jest.restoreAllMocks();
   });
 
-  describe('constructor', () => {
+  describe(AttachmentsStreamingPool.prototype.constructor.name, () => {
```
[the remark this answers is not in the document]
Comment (gasperzgonec) [reply src/attachments-streaming/attachments-streaming-pool.test.ts:69]: I thought we did that [here](https://github.com/devrev/adaas-sdk/blob/main/CONTRIBUTING.md?plain=1#L49).

File: package.json [hunk 10]
```diff
@@ -11,7 +11,7 @@
     "start": "ts-node src/index.ts",
     "lint": "eslint .",
     "lint:fix": "eslint .  --fix",
-    "test": "jest --forceExit --coverage --runInBand"
+    "test": "jest --forceExit --coverage --runInBand --testPathIgnorePatterns='./dist/'"
```
[the remark this answers is not in the document]
Comment (gasperzgonec) [reply package.json:14]: You are correct, this is already present in the `jest.config.cjs`. Now there's multiple ways we could go about doing this, with various cons and pros...
I'd say that the best approach might be that we add things we absolutely don't want in the jest, into the config, but everything else can be in separate test commands, so that the testing can be granular.

[the remark this answers is not in the document]
Comment (radovanjorgic) [reply package.json:14]: Sure, let's do that. Resolve the comment once you add the fix and merge when ready.
